highfidelity/hifi
# Fix VRMenu flashing

Ported `handControllerPointer.js` VR Menu opening at desired position over to C++ to stop it from flashing. 

Added 3 controller actions for different context menu positions:
```js
Actions.ContextMenuCentered // Gamepad start button press
Actions.ContextMenuLeftLaser // Left handcontroller menu button press
Actions.ContextMenuRightLaser // Right handcontroller menu button press
```

## Testplan

This test requires HMD, handcontrollers and an Xbox controller or other supported gamepad

1. Test if the VR menu stopped from flashing.
2. Desktop mode test:
    - Check the gamepad is creating the menu at the cursor position on press of start
    - (optional) If in possession of hydra's try pointing at the 2D screen, press the menu button on the held controller and notice that the menu position depends on your aim direction.
3. HMD mode test:
    - The gamepad should create the menu in the direction that is being looked at press of start button
    - The hand controllers should create the VRMenu in the direction that is pointed at after pressing the controllers menu button.
    - Repeat the actions above a few times in multiple directions, to confirm that the menu positioning algorithm compares with the current release build.
